### DISCUSSION TITLE
Access Function Block under Interface variable

### DISCUSSION THREAD

**Discussion by EnricoTaglini** [stlehmann/pyads#436] [Q&A] [answered]
Hello everyone,

I'm wondering if it's possible to access function block variables not directly from the instance of the FB itself, but through an interface to which the FB is assigned.

For example, I have the following setup:

1. An interface _I_Button_ with a method _Press_.
2. A function block _FB_ToggleButton_ that implements the _I_Button_ interface and contains a variable _IsPressed_.

![image](https://github.com/user-attachments/assets/c9db1367-1d64-46c6-b528-2a9ea1b9dd1b)
![image](https://github.com/user-attachments/assets/51c23ac6-864b-43f8-9483-56c94ab7fa29)

In the main program:

- I declare a new _FB_ToggleButton_ instance (_toggleButton_).
- I also declare an I_Button variable (_myButton_) and assign _toggleButton_ to it.

![image](https://github.com/user-attachments/assets/a89aa68b-7966-4a17-802a-0caa6d711217)

Currently, I can access _IsPressed_ via ADS using the path _MAIN.toggleButton.IsPressed_.

![image](https://github.com/user-attachments/assets/f1850a32-a528-47f6-8160-2ae6655eabff)

Is it possible to access the _IsPressed_ variable through the interface variable _myButton_ instead of directly using the _toggleButton_ instance?

**Comment by chrisbeardy** [comment]
I think I remember reading something about this, let me see if I can dig this out

**Comment by chrisbeardy** [comment, marked as answer]
I just looked into this, unfortunately this is not possible (in pyads or otherwise). ADS itself does not support this, it doesn't expose the inner values over the interface, i.e. the interface's index and offset just point to the "concrete" FB, so ADS does not "resolve" the correct index and offset for the actual variable of interest.

**Reply by EnricoTaglini** [reply]
Got it! Thank you very much for your support!
